core/src/main/java/hudson/triggers/Trigger.java

```diff
@@ -279,9 +280,9 @@ public void run(AbstractProject p) {
         }
 
         // Process all triggers, except SCMTriggers when synchronousPolling is set
-        for (ParameterizedJobMixIn.ParameterizedJob<?, ?> p : inst.allItems(ParameterizedJobMixIn.ParameterizedJob.class)) {
+        for (TriggeredItem p : inst.allItems(TriggeredItem.class)) {
             for (Trigger t : p.getTriggers().values()) {
-                if (!(t instanceof SCMTrigger && scmd.synchronousPolling)) {
+                if (!(p instanceof AbstractProject && t instanceof SCMTrigger && scmd.synchronousPolling)) {
```

Comment:
Not really related, but I noticed that https://github.com/jenkinsci/jenkins/blob/1df554abd76811fd58e6383a614e224700aeecaf/core/src/main/java/hudson/triggers/Trigger.java#L262-L266 did not apply to Pipeline projects, so turning on this flag would have skipped them in SCM polling.

Reply:
Shouldn't the block you referenced be updated to use `TriggeredItem` as well and take a `Job` instead of an `AbstractProject` ?

Reply:
No, it really needs `AbstractProject`: https://issues.jenkins.io/browse/JENKINS-29913